client/app/bundles/users/pages/SignUpLandingPage.tsx

```diff
@@ -0,0 +1,46 @@
+import { Navigate } from 'react-router-dom';
+import { Typography } from '@mui/material';
+
+import Link from 'lib/components/core/Link';
+import { useEmailFromLocationState } from 'lib/containers/AuthPagesContainer';
+import useTranslation from 'lib/hooks/useTranslation';
+
+import Widget from '../components/Widget';
+import translations from '../translations';
+
+const SignUpLandingPage = (): JSX.Element | null => {
+  const { t } = useTranslation();
+
+  const email = useEmailFromLocationState();
+  if (!email) return <Navigate to="/users/sign_up" />;
+
+  return (
+    <Widget
+      subtitle={t(translations.signUpCheckYourEmailSubtitle, {
+        email,
+        strong: (chunk) => <strong>{chunk}</strong>,
+      })}
+      title={t(translations.checkYourEmail)}
+    >
+      <Widget.Foot className="flex space-x-3">
+        <Typography color="text.secondary" variant="body2">
+          {t(translations.confirmedYourEmail)}
+        </Typography>
+
+        <Link to="/users/sign_in">{t(translations.signIn)}</Link>
+      </Widget.Foot>
+
+      <Widget.Foot className="flex space-x-3">
+        <Typography color="text.secondary" variant="body2">
+          {t(translations.didntReceiveConfirmationEmail)}
+        </Typography>
+
+        <Link to="/users/confirmation/new">
```

Comment:
For this one, could it be triggered automatically, stay on the page, and disable resending the confirmation email for another 10 seconds to avoid spamming?

Reply:
What's the point? Someone could just refresh the page and make another request to resend a confirmation email.

I specifically didn't want to do anything about debouncing this unless the solution comes from the server.

Reply:
Yeah we should do server-side debouncing. Lets do this when working on the authentication porting next